### PR TITLE
Feature/TW-977: Bonjour Gateway 

### PR DESCRIPTION
**Pull request by Irtiza-h30**
JIRA: [TW-977](https://connectustechnologies.atlassian.net/browse/TW-977)

## Description
*Summary of this PR
Bonjour Gateway Form*

### Before this PR
*Screenshots of what it looked like before this PR*

### After this PR
*Screenshots of what it will look like after this PR 
image
image
image
image
image
image
image
*

### CODE REVIEW

File: src/containers/ProfileDetails/components/BonjourGateway/index.js [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+                                    parseInt(
+                                      getFieldValue(['bonjourServices', i.name, 'vlanId']),
+                                      10
+                                    )
+                                  );
+
+                                  const index = ids.findIndex(id => id === currentId);
+
+                                  ids.splice(index, 1);
+
```
Comment (sean-macfarlane) [inline src/containers/ProfileDetails/components/BonjourGateway/index.js:150]: `const occurence = ids.filter(item => item === currentId).length;`

File: src/containers/ProfileDetails/components/BonjourGateway/index.js [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                                  if (
+                                    !value ||
+                                    (currentId <= 4095 &&
+                                      currentId > 1 &&
+                                      ids.indexOf(currentId) === -1)
```
Comment (sean-macfarlane) [inline src/containers/ProfileDetails/components/BonjourGateway/index.js:155]: `occurence <= 1`

File: src/containers/ProfileDetails/components/BonjourGateway/index.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+                        initialValue="custom"
+                      >
+                        <Radio.Group size="small" onChange={() => onRadioChange(field.name)}>
+                          <Radio value="custom">Use Custom VLAN</Radio>
+                          <Radio value="default">Use Default VLAN</Radio>
+                        </Radio.Group>
+                      </Item>
+                    ) : (
+                      <div>Use Custom VLAN</div>
+                    )}
```
[the remark this answers is not in the document]
Comment (Irtiza-h30) [reply src/containers/ProfileDetails/components/BonjourGateway/index.js]: Sorry, do you mean `form.getFieldValue(['bonjourServices', field.name, 'vlanIdConfiguration']) === 'default' || !defaultVlanSelected `? It displays it for the specific field if the radio is default, or displays it for all fields if defaultVlanSelected is false (so there is no radio selection with default)

File: src/containers/ProfileDetails/components/BonjourGateway/index.js [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+        });
+      });
+    }
+    return arr;
+  }, [details]);
+
+  useEffect(() => {
+    form.setFieldsValue({
+      profileDescription: details.profileDescription,
+      bonjourServices: bonjourServices.length ? [...bonjourServices] : [''],
```
[the remark this answers is not in the document]
Comment (Irtiza-h30) [reply src/containers/ProfileDetails/components/BonjourGateway/index.js:36]:
```
      bonjourServices: details?.bonjourServices?.map(i => {
        return {
          vlanId: i.vlanId,
          vlanIdConfiguration: i.vlanId ? 'custom' : 'default',
          supportAllServices: i.supportAllServices.toString(),
          serviceNames: i.serviceNames?.filter(service => service != null),
        };
      }) || [''],
```
I had to change it to this. The filter for the serviceNames ensures that there's not a default null value when the user toggles from 'All Services' to 'Selected Services' for a specific field. And the empty array is so there is initial fields for when the user wants to add a new bonjour service

File: src/containers/ProfileDetails/components/BonjourGateway/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sean-macfarlane) [reply src/containers/ProfileDetails/components/BonjourGateway/index.js]: `form.getFieldValue(['bonjourServices', field.name, 'vlanIdConfiguration']) !== 'default' && defaultVlanSelected ? <div /> : <Item />`

**Pull request merged**